CodingTrackerLibrary/Controller.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Dapper;
+using System.Data.SQLite;
+
+namespace CodingTracker
+{
+    
+    public class Controller
+    {
+        public static void CreateDatabase(string connectionString)
+        {
+            try
+            {
+                using (var conn = new SQLiteConnection(connectionString))
+                {
+                    conn.Open();
+                    string createTableQuery = "CREATE TABLE IF NOT EXISTS records ( Id INTEGER PRIMARY KEY AUTOINCREMENT, DateStart TEXT, DateEnd TEXT, Duration TEXT )";
+                    conn.Execute(createTableQuery);
+                }
+            }
+            catch (SQLiteException ex) { Console.WriteLine($"SQLiteException Error: {ex.Message}"); }
+            catch (Exception ex) { Console.WriteLine($"Error: {ex.Message}"); }
+        }
+
+        public static void InsertData(Record CodingRecord, string connectionString)
+        {
+            try
+            {
+                using (var conn = new SQLiteConnection(connectionString))
+                {
+                    conn.Open();
+                    string insertQuery = "INSERT INTO records (DateStart, DateEnd, Duration) VALUES(@DateStart, @DateEnd, @Duration)";
+                    conn.Query(insertQuery, CodingRecord);
+                }
+            }
+            catch (SQLiteException ex) { Console.WriteLine($"SQLiteException Error: {ex.Message}"); }
+            catch (Exception ex) { Console.WriteLine($"Error: {ex.Message}"); }
+        }
+        public static IEnumerable<dynamic> ReadData(string connectionString)
```

Comment:
According to the requirements you cannot return dynamic object when reading from database.
It should return
IEnumarable<Model> or List<Model> (in your case)

"When reading from the database, you can't use an anonymous object, you have to read your table into a List of Coding Sessions."